Adaptors/Amqp/src/PullQueueStorage.cs

```diff
@@ -69,46 +65,52 @@ public override async Task Init(CancellationToken cancellationToken)
       await ConnectionAmqp.Init(cancellationToken)
                           .ConfigureAwait(false);
 
-      receivers_ = Enumerable.Range(0,
-                                    NbLinks)
-                             .Select(i => CreateReceiver(ConnectionAmqp,
-                                                         i))
-                             .ToArray();
-
-      senders_ = Enumerable.Range(0,
-                                  NbLinks)
-                           .Select(i => new AsyncLazy<ISenderLink>(async () => new SenderLink(new Session(await ConnectionAmqp.GetConnectionAsync()
-                                                                                                                              .ConfigureAwait(false)),
-                                                                                              $"{Options.PartitionId}###SenderLink{i}",
-                                                                                              $"{Options.PartitionId}###q{i}")))
-                           .ToArray();
-
-      var senders = senders_.Select(lazy => lazy.Value)
-                            .WhenAll();
-      var receivers = receivers_.Select(lazy => lazy.Value)
-                                .WhenAll();
-      await Task.WhenAll(senders,
-                         receivers)
-                .ConfigureAwait(false);
       IsInitialized = true;
     }
   }
 
   /// <inheritdoc />
-  public async IAsyncEnumerable<IQueueMessageHandler> PullMessagesAsync(int                                        nbMessages,
+  public async IAsyncEnumerable<IQueueMessageHandler> PullMessagesAsync(string                                     partitionId,
+                                                                        int                                        nbMessages,
                                                                         [EnumeratorCancellation] CancellationToken cancellationToken = default)
   {
     var nbPulledMessage = 0;
 
+    if (string.IsNullOrEmpty(partitionId))
+    {
+      throw new ArgumentException($"{nameof(partitionId)} is not defined.");
+    }
+
```

Comment:
not needed here. It should be added in the Pollster so that it is not needed in every plugin.